test/cli/apps/list.test.js

```diff
@@ -52,21 +64,108 @@ describe('eg apps list', () => {
         };
         generator.stdout = message => {
           let app = JSON.parse(message);
-          output[app.name] = true;
-          output[app.id] = true;
+          output[app.id] = app;
+        };
+      });
+
+      generator.once('end', () => {
+        assert.equal(_.size(output), apps.length);
+        for (const app of apps) {
+          assert.ok(output[app.id]);
+          assert.ok(output[app.id].name, app.name);
+        }
+        done();
+      });
+    });
+
+    env.argv = program.parse('apps list');
+  });
+
+  it('should show apps with specific name', done => {
+    env.hijack(namespace, generator => {
+      let output = {};
+
+      generator.once('run', () => {
+        generator.log.error = message => {
+          done(new Error(message));
+        };
+        generator.stdout = message => {
+          let app = JSON.parse(message);
+          output[app.id] = app;
+        };
+      });
+
+      generator.once('end', () => {
+        const filteredApps = apps.filter((app) => app.name === 'app1');
+        assert.equal(_.size(output), filteredApps.length);
+        for (const app of filteredApps) {
+          assert.ok(output[app.id]);
+          assert.ok(output[app.id].name, app.name);
+        }
+        done();
+      });
+    });
+
+    env.argv = program.parse('apps list -n app1');
+  });
+
+  it('should show user apps', done => {
+    env.hijack(namespace, generator => {
+      let output = {};
+
+      generator.once('run', () => {
+        generator.log.error = message => {
+          done(new Error(message));
+        };
+        generator.stdout = message => {
+          let app = JSON.parse(message);
+          output[app.id] = app;
+        };
+      });
+
+      generator.once('end', () => {
+        const filteredApps = apps.filter((app) => app.userId === user1.id);
+        assert.equal(_.size(output), filteredApps.length);
+        for (const app of filteredApps) {
+          assert.ok(output[app.id]);
+          assert.ok(output[app.id].name, app.name);
+        }
+        done();
+      });
+    });
+
+    env.argv = program.parse(`apps list -u ${user1.id}`);
+  });
+
+  it('should show user apps with specific name', done => {
+    env.hijack(namespace, generator => {
+      let output = {};
+
+      generator.once('run', () => {
+        generator.log.error = message => {
+          done(new Error(message));
+        };
+        generator.stdout = message => {
+          let app = JSON.parse(message);
+          output[app.id] = app;
         };
       });
 
       generator.once('end', () => {
-        assert.ok(output[app1.name]);
-        assert.ok(output[app1.id]);
-        assert.ok(output[app2.name]);
-        assert.ok(output[app2.id]);
+        const filteredApps = apps.filter((app) => (
+          app.userId === user1.id &&
+          app.name === 'app1'
+        ));
+        assert.equal(_.size(output), filteredApps.length);
```

Comment:
How it is possible to have for the same user app with the same name ? 


and in general `_.size(output)` could be Object.keys(output).length with no dependencies 
https://github.com/lodash/lodash/blob/master/size.js#L40

Reply:
@DrMegavolt the question is should we use lodash ...

Reply:
I believe situation is the same as bluebird. there is only one function (_.omit) usage across the code (3 places) that has no native analog. and can be replaced as small module - https://www.npmjs.com/package/lodash.omit

and some weird ` _.map` in test usage that can be just changed to standard map\for..in\for..of

Reply:
update: lodash.omit module is just an export from ldoash. still has lodash as dependency, but the idea is still the same - it can be replaced